### ISSUE TITLE
School Bells

### ISSUE THREAD

**Issue by hburroughsSJE** [floft/feedback#2]
Floft,

My name is Henry Burroughs, and I am the Director of Technology at a school in Tampa, FL.   We are having to redo our Bell system, and I was going to build a raspberry Pi based bell system where we have a Pi in every classroom attached to an external speaker and play a sound (at the moment a bell sound).   I was doing a Google search one last time before I write my own program, and I stumbled across your program.  It looks great, and the only thing I would need is instead of writing to a serial port, is to run a command (that way you don't have to hardcode a specific play audio file command).

Unfortunately, at this campus we don't already have a pre-wired bell system.   Bells were previously run on the PCs using a program called TimeChimes (which made it a pain to update the schedules).   Running the PIs I could at least mass update the xml file across all the units.  Plus having the PIs allows me to turn them into iBeacons with a USB dongle.  My C++ is a bit rusty (anything I write currently is Perl or Bash).  My guess is we could have a option in the XML that would indicate if it was a serial port or a command to run.   Then the user could fill in the field with the command to be run.

Please let me know if there is anything I could do to help make this possible.  Hope you are having a nice summer.

**Comment by floft** [comment]
Indeed, adding the ability to run a command to the daemon is definitely possible, and shouldn't take much work. Maybe something like this, specifying which method to use in `method` rather than as a `gpio` boolean.
````
<settings>
    ...
    <method>gpio|command|serial</method>
    <command>path/to/something arg1 arg2</command>
</settings>
````

I don't have time to add it at the moment, but I'll see if I can look into doing that soon.

**Comment by floft** [comment]
What do you think about this? https://github.com/floft/BellSystem/commit/dd3d570389b374cc50a824be599c8445e9404d08

**Comment by hburroughsSJE** [comment]
That looks great!  I like how you disabled running a command from the web interface for the moment due to security.   I haven’t had a time to compile the program yet on a Pi, but I will let you know how it goes.  I’m out on vacation next week, but after that I should be able to give it a spin.   I take it the init scripts are running it as a daemon root/superuser at the moment?




Henry


Henry B. Burroughs, IIIDirector of Technology

St. John’s Episcopal Parish Day School

www.stjohnseagles.org

**Comment by floft** [comment]
Yeah, presently the way I set it up runs it as root. Although, I was actually using SysVinit rather than systemd, so when I eventually write systemd service files, I can just add something like User=bellsystem.

**Comment by floft** [comment]
Feel free to re-open this if you have further comments.

**Comment by hburroughsSJE** [comment]
Garrett,

You can close it back, but I just wanted to say Thank you.  It was exactly
the right piece of software I needed at the right time.   I love open
source.  It has been working great.

God Bless,

Henry

On Mon, Sep 7, 2015 at 8:02 PM, Garrett Wilson <notifications@github.com>
wrote:

> Closed #2 <https://github.com/floft/feedback/issues/2>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/floft/feedback/issues/2#event-403098830>.
>



-- 
Henry B. Burroughs, III
Director of Technology
St. John's Episcopal Parish Day School
www.stjohnseagles.org
